Compare against expected output for formatting tests

natecook1000 opened the pull request:
The formatting request tests currently test for the specific list of changes returned by the server, but that list is tied to the differences algorithm used by the standard library, which isn't guaranteed to be stable. This change compares the expected result with a new string generated by applying the list of changes, which is stable.

_Note:_ This includes a function for applying the list of `TextEdit`s received from the server to a string, since I couldn't find an existing method with that purpose.

natecook1000 commented:
This change should enable swiftlang/swift#83212 to land.

bnbarham commented on Sources/SKTestSupport/String+applyingTextEdits.swift:
```diff
@@ -0,0 +1,65 @@
+  /// Returns a new string obtained by applying the given text edits to this
+  /// string.
+  package func applying(_ edits: [TextEdit]) -> String {
+    let sortedEdits = edits.sorted { lhs, rhs in
+      lhs.range.lowerBound < rhs.range.lowerBound
+    }
```
I would generally expect this to already be the case, is it not with the new algorithm? We should just be able to go through this list back to front, applying the changes along the way. `RenameAssertions.swift` has a private function that does that already (`apply(edits:to:)`) - we could move that into `SKTestSupport` instead and use it here?

natecook1000 commented on Tests/SourceKitLSPTests/FormattingTests.swift:
```diff
@@ -25,15 +25,13 @@ final class FormattingTests: XCTestCase {
-      """
+    let source = """
       struct S {
       1️⃣var foo: 2️⃣ 3️⃣Int
```
🤦🏼‍♂️ Of course!

natecook1000 commented:
@bnbarham Thanks for the notes! I removed my new function for applying changes and switched to the one in `RenameAssertions.swift`, and took out the markers where they are no longer needed.
